addons/im_livechat/static/src/embed/common/close_confirmation.js

```diff
@@ -1,12 +1,18 @@
-import { Component } from "@odoo/owl";
-import { useAutofocus } from "@web/core/utils/hooks";
+import { Component, onMounted, onWillUnmount, useRef } from "@odoo/owl";
 
 export class CloseConfirmation extends Component {
     static template = "im_livechat.CloseConfirmation";
     static props = ["onCloseConfirmationDialog", "onClickLeaveConversation"];
 
     setup() {
-        useAutofocus({ refName: "root" });
+        this.root = useRef("root");
+        onMounted(() => this.root.el.focus());
+        onWillUnmount(() =>
+            this.root.el
+                .closest(".o-mail-ChatWindow")
+                .querySelector(".o-mail-Composer-input")
+                .focus()
+        );
```

Comment:
Prefer letting `ChatWindow` handle the focus on composer input itself, with `onCloseConfirmationDialog`.

The `CloseConfirm` component should not have chat window concern. This is also prone to mis-focusing the wrong chat window...